File: LeetCode/Longest Subarray with Maximum Bitwise AND.py
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        max_element = max(nums)
        mx = cnt = 0
        
        for num in nums:
            if num == max_element:
                cnt+=1
            else:
                cnt=0
            
            mx = max(mx, cnt)
        
        return mx

File: LeetCode/Longest Subarray with Maximum Bitwise AND.cpp
class Solution {
public:
    int longestSubarray(vector<int>& nums) {
    int num = *max_element(nums.begin(), nums.end());
    int mx = 0, cnt = 0;
    
    for(int i=0; i < nums.size(); i++) {
        if (nums[i] == num)
            cnt++;
        else 
            cnt = 0;
        
        mx = max(mx, cnt);
    }
    
    return mx;
    }
};